src/RecipeDetail.tsx

```diff
@@ -0,0 +1,87 @@
+import { useEffect, useState } from 'react';
+import { useParams } from 'react-router-dom';
+
+import styles from './RecipeDetail.module.css';
+type Recipe = {
+  id: number;
+  name: string;
+  difficulty: string;
+  tags: string[];
+  image: string;
+  ingredients: string[];
+  instructions: string[];
+  prepTimeMinutes: number;
+  cookTimeMinutes: number;
+  cuisine: string;
+  mealType: string[];
+  caloriesPerServing: number;
+  rating: number;
+  reviewCount: number;
+  servings: number;
+  userId: number;
+};
+
+const RecipeDetail = () => {
+  const { id } = useParams<{ id: string }>();
+  const [recipe, setRecipe] = useState<Recipe | null>(null);
+  const [error, setError] = useState('');
+
+  useEffect(() => {
+    fetch(`https://dummyjson.com/recipes/${id}`)
+      .then((res) => res.json())
+      .then((data) => setRecipe(data as Recipe))
+      .catch(() => setError('레시피 정보를 불러오는데 실패했습니다.'));
+  }, [id]);
+
+  if (error) return <p>{error}</p>;
+  if (!recipe) return;
```

Comment:
null 값이 falsy하기 때문에 해당 구문이 동작은 할 것 같은데, 이왕이면 아래와 같이 명시적으로 작성해주시면 좋을 것 같습니다. 또한 컴포넌트에서는 void return을 하면 에러가 발생할 수 있고, 공식 문서에서도 null 값을 리턴하고 있습니다. (return false를 사용하는 경우도 있기는 한데, return null이 더 일반적인 것 같아요)
https://react.dev/learn/conditional-rendering#conditionally-returning-nothing-with-null

```suggestion
  if (recipe === null) return null;
```

Reply:
UX를 위해서 로딩 상태도 처리해주면 어떨까요??